portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,40 +57,72 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String username = request.getParameter("username");
     String comment = request.getParameter("new-comment");
     String inputNumber = request.getParameter("input-number");
+    String email;
+    if(userService.isUserLoggedIn()){
+      email = userService.getCurrentUser().getEmail();
+    } else {
+      email = "null";
+    }
     if (inputNumber != null) {
       numComments = Integer.parseInt(inputNumber);
     }
     if (comment != null) {
-      storeComment(username, comment);
+      storeComment(username, comment, email);
     }
     response.sendRedirect("/index.html");
   }
 
   /** Converts the comments Array to json using gson */
-  private String convertToJson(ArrayList<String[]> comments) {
+  private String convertToJson(ArrayList<String> comments) {
     Gson gson = new Gson();
     String json = gson.toJson(comments);
     return json;
   }
 
+  private String finalJson(ArrayList<String> comments){
```

Comment:
Might be helpful to explain what this function and convertCommentToJson do, perhaps include an example of what it looks like when formatted